Doc/library/typing.rst

```diff
@@ -1584,16 +1584,31 @@ These are not used in annotations. They are building blocks for creating generic
 
       assert isinstance(open('/some/file'), Closable)
 
+      @runtime_checkable
+      class HasErrNo(Protocol):
+          errno: int
+
+      assert isinstance(OSError(42, 'foo'), HasErrNo)
+
    .. note::
 
-        :func:`runtime_checkable` will check only the presence of the required
-        methods, not their type signatures. For example, :class:`ssl.SSLObject`
+        :func:`!runtime_checkable` will check only the presence of the required
+        methods or attributes, not their type signatures or types.
+        For example, :class:`ssl.SSLObject`
         is a class, therefore it passes an :func:`issubclass`
         check against :data:`Callable`.  However, the
         ``ssl.SSLObject.__init__`` method exists only to raise a
         :exc:`TypeError` with a more informative message, therefore making
         it impossible to call (instantiate) :class:`ssl.SSLObject`.
 
+   .. note::
+
+        An :func:`isinstance` check against a runtime-checkable protocol can be
+        surprisingly slow compared to an ``isinstance()`` check against
+        a non-protocol class. Consider using alternative idioms such as
```

Comment:
Great as is, but if we want to sow further doubt:
```suggestion
        a non-protocol class (and very difficult to spot on a profile). Consider using alternative idioms such as
```

Reply:
Hmm, I'm inclined to leave this one as is — not sure it's worth the added verbosity, and I want to get on as quickly as possible to "here's what to do instead"